src/App.tsx

```diff
@@ -15,48 +15,117 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[], sortType:SortType, isReversed:boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1: string, good2: string) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortAlphabetically = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const sortReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const reset = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
+  const visGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info 
+          ${sortType === SortType.ALPHABET ? '' : 'is-light'}`}
+          onClick={sortAlphabetically}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={`button is-success  
+          ${sortType === SortType.LENGTH ? '' : 'is-light'}`}
+          onClick={sortByLength}
+
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={`button is-warning 
+          ${isReversed ? '' : 'is-light'}`}
+          onClick={sortReverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {sortType !== SortType.NONE || isReversed !== false
+          ? (
+            <button
+              type="button"
+              data-cy="button"
+              className="button is-danger is-light "
+              onClick={reset}
+            >
+              Reset
+            </button>
+          )
+          : ''}
```

Comment:
```suggestion
          && (
            <button
              type="button"
              data-cy="button"
              className="button is-danger is-light "
              onClick={reset}
            >
              Reset
            </button>
          )}
```
You can use logical && instead of the ternary operator if you only need to render the button or not
https://react.dev/learn/conditional-rendering#logical-and-operator-